adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,40 @@
 #include "impl.h"
 #include <vector>
+#include <tuple>
+#include <stack>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+std::pair<char, char> get_matching_brackets_pair(char c) {
+	for (auto &bracket_pair: brackets) {
+		if (c == bracket_pair.first || c == bracket_pair.second)
+			return bracket_pair;
+		}
+	return {};
+}
+
+
+bool isValid(const std::string& source){
+	std::stack<char> stack;
+	for (auto c : source){
+		char openBracket, closingBracket;
+		std::tie(openBracket, closingBracket) = get_matching_brackets_pair(c);
+		if (openBracket != closingBracket)
```

Comment:
if (const bool isBracket = (openBracket != closingBracket) is much readable, name your conditions!